dodiscover/constraint/skeleton.py

```diff
@@ -0,0 +1,470 @@
+import logging
+from collections import defaultdict
+from itertools import combinations
+from typing import Any, Dict, Iterable, List, Set
+
+import networkx as nx
+import numpy as np
+
+from dodiscover.ci import BaseConditionalIndependenceTest
+
+from ..context import Context
+
+logger = logging.getLogger()
+
+ACCEPTED_MARKOVIAN_SKELETON_METHODS = ["complete", "neighbors", "neighbors_path"]
+
+
+def _iter_conditioning_set(
+    possible_variables: Iterable,
+    x_var: Any,
+    y_var: Any,
+    size_cond_set: int,
+):
+    """Iterate function to generate the conditioning set.
+
+    Parameters
+    ----------
+    possible_variables : iterable
+        A set/list/dict of possible variables to consider for the conditioning set.
+        This can be for example, the current adjacencies.
+    x_var : node
+        The node for the 'x' variable.
+    y_var : node
+        The node for the 'y' variable.
+    size_cond_set : int
+        The size of the conditioning set to consider. If there are
+        less adjacent variables than this number, then all variables will be in the
+        conditioning set.
+
+    Yields
+    ------
+    Z : set
+        The set of variables for the conditioning set.
+    """
+    exclusion_set = {x_var, y_var}
+
+    all_adj_excl_current = [p for p in possible_variables if p not in exclusion_set]
+
+    # loop through all possible combinations of the conditioning set size
+    for cond in combinations(all_adj_excl_current, size_cond_set):
+        cond_set = set(cond)
+        yield cond_set
+
+
+def _find_neighbors_along_path(G: nx.Graph, start, end) -> Set:
+    def _assign_weight(u, v, edge_attr):
+        if u == node or v == node:
+            return np.inf
+        else:
+            return 1
+
+    nghbrs = set()
```

Comment:
Just a minor thing, but sometimes it might be better to be more descriptive about the parameter names. Here, instead of ```nghbrs``` simply write it out ```neighbors```. Its not that much longer, but looks much less cryptic.

Reply:
WDYT about using networkx partners? so instead something like `nbr` so it's shorter and not prone to misspelling?

ref: https://github.com/networkx/networkx/pull/5898#discussion_r937274643